crates/radix-engine-toolkit/src/utils.rs

```diff
@@ -43,80 +43,93 @@ pub fn manifest_from_intent(intent: &IntentV1) -> TransactionManifestV1 {
 }
 
 pub fn network_definition_from_network_id(network_id: u8) -> NetworkDefinition {
+    network_definition_from_network_id_strict(network_id).unwrap_or(
+        NetworkDefinition {
+            id: 0x25,
+            logical_name: "unnamed".to_string(),
+            hrp_suffix: format!("tdx_{:x}_", network_id),
+        },
+    )
+}
+
+pub fn network_definition_from_network_id_strict(
```

Comment:
In my opinion, this defeats the purpose of the network id being an integer as its functioning like an enum essentially. But the network id understood by the protocol is just an integer.